### PR TITLE
Update to fix ArmResourceExtensionClient

### PR DESCRIPTION
**Pull request by HarveyLink**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: test/TestProjects/MgmtScopeResource/Customized/CustomizedArmResourceExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+using Azure.Core;
+using Azure.ResourceManager.Core;
+
+[assembly:CodeGenSuppressType("ModelToBeSkipped")]
+[assembly:CodeGenSuppressType("EnumToBeSkipped")]
+[assembly:CodeGenSuppressType("EnumToBeSkippedExtensions")]
+
+namespace MgmtScopeResource
+{
+    [CodeGenSuppress("GetPolicyAssignments", typeof(PolicyAssignmentCollection))]
```
Comment (fengzhou-msft) [inline test/TestProjects/MgmtScopeResource/Customized/CustomizedArmResourceExtensions.cs:15]: Try this:
```suggestion
    [CodeGenSuppress("GetPolicyAssignments", typeof(ArmResource))]
```

File: src/AutoRest.CSharp/Mgmt/AutoRest/MgmtTarget.cs [hunk 2]
```diff
@@ -146,10 +146,11 @@ public static void Execute(GeneratedCodeWorkspace project, CodeModel codeModel,
...
             {
-                var armResourceExt = MgmtContext.Library.ArmResourceExtensions;
-                var armResourceExtensionsCodeWriter = new ArmResourceExtensionsWriter(armResourceExt);
-                armResourceExtensionsCodeWriter.Write();
-                AddGeneratedFile(project, $"Extensions/{armResourceExtensionsCodeWriter.FileName}.cs", armResourceExtensionsCodeWriter.ToString());
+                WriteExtensionPair(project, MgmtContext.Library.ArmResourceExtensionsClient);
+                //var armResourceExt = MgmtContext.Library.ArmResourceExtensions;
+                //var armResourceExtensionsCodeWriter = new ArmResourceExtensionsWriter(armResourceExt);
+                //armResourceExtensionsCodeWriter.Write();
+                //AddGeneratedFile(project, $"Extensions/{armResourceExtensionsCodeWriter.FileName}.cs", armResourceExtensionsCodeWriter.ToString());
```
Comment (ArcturusZhang) [inline src/AutoRest.CSharp/Mgmt/AutoRest/MgmtTarget.cs:153]: We should clean these commented code

File: test/AutoRest.TestServer.Tests/Mgmt/TestProjects/MgmtScopeResourceTests.cs [hunk 3]
```diff
@@ -16,7 +16,7 @@ public class MgmtScopeResourceTests : TestProjectTests
         [TestCase("ManagementGroupExtensions", "GetPolicyAssignments", false)]
         [TestCase("SubscriptionExtensions", "GetPolicyAssignments", false)]
         [TestCase("ResourceGroupExtensions", "GetPolicyAssignments", false)]
-        [TestCase("ArmResourceExtensions", "GetPolicyAssignments", true)]
+        //[TestCase("ArmResourceExtensions", "GetPolicyAssignments", true)]   --Get from Azure.Core
```
Comment (ArcturusZhang) [inline test/AutoRest.TestServer.Tests/Mgmt/TestProjects/MgmtScopeResourceTests.cs:19]: This should be removed

File: test/TestProjects/MgmtScopeResource/Customized/CustomizedArmResourceExtensions.cs (hunk 1 above)
Comment (HarveyLink) [reply test/TestProjects/MgmtScopeResource/Customized/CustomizedArmResourceExtensions.cs]: Done

File: src/AutoRest.CSharp/Mgmt/AutoRest/MgmtTarget.cs (hunk 2 above)
Comment (HarveyLink) [reply src/AutoRest.CSharp/Mgmt/AutoRest/MgmtTarget.cs]: Done

File: test/AutoRest.TestServer.Tests/Mgmt/TestProjects/MgmtScopeResourceTests.cs (hunk 3 above)
Comment (HarveyLink) [reply test/AutoRest.TestServer.Tests/Mgmt/TestProjects/MgmtScopeResourceTests.cs]: Done

File: test/TestProjects/MgmtScopeResource/Generated/Extensions/ArmResourceExtensions.cs [hunk 4]
```diff
@@ -12,14 +12,11 @@ namespace MgmtScopeResource
     /// <summary> A class to add extension methods to ArmResource. </summary>
     public static partial class ArmResourceExtensions
     {
-        /// <summary> Gets a collection of PolicyAssignments in the PolicyAssignment. </summary>
-        /// <param name="armResource"> The <see cref="ArmResource" /> instance the method will execute against. </param>
-        /// <returns> An object representing collection of PolicyAssignments and their operations over a PolicyAssignment. </returns>
-        public static PolicyAssignmentCollection GetPolicyAssignments(this ArmResource armResource)
```
Comment (m-nash) [inline test/TestProjects/MgmtScopeResource/Generated/Extensions/ArmResourceExtensions.cs:18]: What happened to the GetPolicyAssignments method?

Comment (HarveyLink) [reply test/TestProjects/MgmtScopeResource/Generated/Extensions/ArmResourceExtensions.cs:18]: As discussed, changed name to FakePolicyAssignments

**Review by m-nash — APPROVED** [review]

**Pull request merged**